apps/zbugs/shared/queries.ts

```diff
@@ -42,39 +42,63 @@ const issueRowSort = z.object({
 
 type IssueRowSort = z.infer<typeof issueRowSort>;
 
+const defineAuthQuery = defineQueryWithContextType<AuthData | undefined>();
+
 export const queries = {
-  allLabels: syncedQuery('allLabels', z.tuple([]), () => builder.label),
+  allLabels: defineQuery(
+    'allLabels',
+    {
+      validator: z.undefined(),
+    },
+    () => builder.label,
+  ),
 
-  allUsers: syncedQuery('allUsers', z.tuple([]), () => builder.user),
+  allUsers: defineQuery(
```

Comment:
When I wrote the doc I wrote it with the queries colocated because a lot of people had asked for that.

After using it myself a lot I still think that the predominate way to use Zero will be to put the queries together in a central file. As the app grows, there will likely be per-area files. But it ends up being convenient to see all the queries at a glance since they are an expensive resource you are trying to manage.

For this reason maybe Matt's original formulation of:

```ts
defineQuery({
  allLabels: ({validator: z.string()}) => {

  }
});
```

makes more sense